File: distance_measurement.js
const pigpio = require('pigpio-client')().then((pi) => {
  const TRIGEVEN = 29;
  const ECHOEVEN = 31;
  const TRIGODD = 33;
  const ECHOODD = 35;

  pi.gpio(TRIGEVEN).modeSet('output');
  pi.gpio(ECHOEVEN).modeSet('input');

  console.log('Distance measurement is in progress...');

  pi.gpio(TRIGEVEN).digitalWrite(0);
  console.log('Sensor settles');
  setTimeout(() => {
    pi.gpio(TRIGEVEN).digitalWrite(1);
    setTimeout(() => {
      pi.gpio(TRIGEVEN).digitalWrite(0);

      let pulseStart;
      while (pi.gpio(ECHOEVEN).digitalRead() === 0) {
        pulseStart = process.hrtime.bigint();
      }

      let pulseEnd;
      while (pi.gpio(ECHOEVEN).digitalRead() === 1) {
        pulseEnd = process.hrtime.bigint();
      }

      const pulseDuration = Number(pulseEnd - pulseStart) / 1e6;
      const distance = pulseDuration * 17150;
      console.log(`Distance: ${distance.toFixed(2)} cm`);

      pi.end();
    }, 10);
  }, 2000);
});

File: distance_measurement.py
import RPi.GPIO as GPIO
import time 


GPIO.setmode(gpio.BCM)
GPIO.setwarnings(False)
TRIG = 23
ECHO = 24
print("Distance mesurement is in progress...")
GPIO.setup(TRIG, GPIO.OUT)
GPIO.setup(ECHO, GPIO.IN)
GPIO.output(TRIG, False)
print("Sensor settles")
time.sleep(2)
GPIO.output(TRIG, True)
time.sleep(0.00001)
GPIO.output(TRIG, False)
while GPIO.input(ECHO) == 0:
	pulse_start = time.time()
while GPIO.input(ECHO) == 1:
	pulse_end = time.time()
pulse_duration = pulse_end - pulse_start
distance = pulse_duration * 17150
distance = round(distance, 2)
print("Distance: ",discance," cm")
